app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt

```diff
@@ -0,0 +1,124 @@
+package org.p2p.wallet.auth.ui.restore.common
+
+import android.os.Bundle
+import android.view.View
+import androidx.activity.addCallback
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.view.isVisible
+import org.koin.android.ext.android.inject
+import org.koin.core.parameter.parametersOf
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.ui.phone.PhoneNumberEnterFragment
+import org.p2p.wallet.auth.web3authsdk.GoogleSignInHelper
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentCommonRestoreBinding
+import org.p2p.wallet.restore.ui.keys.SecretKeyFragment
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+class CommonRestoreFragment :
+    BaseMvpFragment<CommonRestoreContract.View, CommonRestoreContract.Presenter>(
+        R.layout.fragment_common_restore
+    ),
+    CommonRestoreContract.View,
+    GoogleSignInHelper.GoogleSignInErrorHandler {
+
+    companion object {
+        fun create(): CommonRestoreFragment = CommonRestoreFragment()
+    }
+
+    override val presenter: CommonRestoreContract.Presenter by inject { parametersOf(this) }
+
+    private val binding: FragmentCommonRestoreBinding by viewBinding()
+
+    private val signInHelper: GoogleSignInHelper by inject()
+
+    private val googleSignInLauncher = registerForActivityResult(
+        ActivityResultContracts.StartIntentSenderForResult(),
+        ::handleSignResult
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            toolbar.setNavigationOnClickListener {
+                popBackStack()
+            }
+            buttonRestoreByGoogle.setOnClickListener {
+                presenter.useGoogleAccount()
+            }
+
+            buttonPhone.setOnClickListener {
+                replaceFragment(PhoneNumberEnterFragment.create())
+            }
+
+            buttonSeed.setOnClickListener {
+                // TODO make a real restore implementation!
+                replaceFragment(SecretKeyFragment.create())
+            }
+        }
+
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
+            popBackStack()
+        }
+    }
+
+    override fun startGoogleFlow() {
+        signInHelper.showSignInDialog(requireContext(), googleSignInLauncher)
+    }
+
+    override fun showError(error: String) {
```

Comment:
you show have similar method in BaseMvpView

Reply:
yes you are right but we need logic 
setLoadingState(isScreenLoading = false) to set in a view on error